### PR TITLE
feat(chart): expose .securityContext.runAsGroup

### LINKED ISSUE #742: Add securityContext configuration to Helm Chart Values file, so runAsUser and runAsGroup can be set for coreDNS
### Is your feature request related to a problem?

No

### Which solution do you suggest?

I may be misunderstanding how vcluster Helm Chart works, but currently, it seems it's not possible to set securityContext at both container and pod levels, so I can't set its runAsUser, runAsGroup, and fsGroup for coreDNS.

This is currently what I see when I <kubectl get pod coredns -o yaml ...>

```
securityContext:
  allowPrivilegeEscalation: false
  capabilities:
    drop:
    - all
  readOnlyRootFilesystem: true
```

...

securityContext: {}

and this is what I see from coreDNS' manifest file:

```
        securityContext:
          runAsUser: {{ .RUN_AS_USER }}
          runAsNonRoot: {{ .RUN_AS_NON_ROOT }}
          allowPrivilegeEscalation: false
          capabilities:
            drop:
            - all
          readOnlyRootFilesystem: true
```

...

```
      {{- if .Values.isolation.enabled }}
      securityContext:
        seccompProfile:
          type: RuntimeDefault
      {{- end }}
```

I'd like to set both runAsUser and runAsUser to 10001 for example through either Values.coredns.securityContext or Values.coredns.podSecurityContext.

[https://kubernetes.io/docs/tasks/configure-pod-container/security-context/](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/)

Thanks in advance!

### Which alternative solutions exist?

*No response*

### Additional context

*No response*

### PR DESCRIPTION
**Pull request by rohantmp**
- fix(chart): ignoring .securityContext values for coredns
- feat(chart): expose .securityContext.runAsGroup for coredns

**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind enhancement

**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible) 
resolves #742 
resolves ENG-414

**Please provide a short message that should be published in the vcluster release notes**

fix: pass securityContext.runAsGroup setting to coredns

**What else do we need to know?** 

Updated documentation as well

### CODE REVIEW

File: pkg/coredns/coredns.go [hunk 1]
```diff
@@ -20,8 +20,10 @@ const (
 	VarImage              = "IMAGE"
 	VarRunAsUser          = "RUN_AS_USER"
 	VarRunAsNonRoot       = "RUN_AS_NON_ROOT"
```
Comment (matskiv) [inline pkg/coredns/coredns.go:22]: nit-pick: `VarRunAsNonRoot` can be deleted now that it is no longer used

**Comment by matskiv** [comment]
Looks good! I am happy to merge once Lint issues are resolved :)

**Comment by rohantmp** [comment]
Lint passed, accidentally named some consts in all caps :laughing:

**Review by matskiv — APPROVED** [review]
Looks good! :+1:

**Pull request merged**